Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -224,21 +224,12 @@ public Vector3 GrabPoint
             {
                 get
                 {
-                    if (IsNearPointer())
-                    {
-                        return pointer.Position;
-                    }
-                    else
-                    {
-                        return (pointer.Rotation * initialGrabPointInPointer) + pointer.Position;
-                    }
+                    return (pointer.Rotation * initialGrabPointInPointer) + pointer.Position;
```

Comment:
Could you please explain why you removed this? It seems like this is changing behavior of GrabPoint so I want to understand repercussions.

Reply:
Yeah, my changes require a distinction between the pointer position, the grab position and the object position:

- pointer position: A point on the hand/controller defined by the type of pointer you're using
- grab position: A point on the object where the pointer grabbed the object
- object position: The origin point of the object

The way this was before, for near interaction the pointer position and the grab position would be equal. In fact, for near interaction pointer.Position returns the palm position. Now, the grab position is where you grabbed the object.

Reply:
### Difference in old(![#ff00ff](https://placehold.it/15/ff00ff/000000?text=+))  vs. new (![#00ffff](https://placehold.it/15/00ffff/000000?text=+)) grab points:
![image](https://user-images.githubusercontent.com/47415945/61887774-2c4fe000-aefa-11e9-9844-3ad39be626ae.png)

Reply:
>for near interaction pointer.Position returns the palm position

Huh, interesting. I thought we moved the grab pointer to ~a point between your index finger and your thumb~ actually it looks like we just put it right at the position of the index finger? Via:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/3c98d9c582e90214fa4685dc02ae78553c36c492/Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs#L87

and

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/3c98d9c582e90214fa4685dc02ae78553c36c492/Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/SpherePointer.cs#L120

![image](https://user-images.githubusercontent.com/3580640/61920724-68417000-af0f-11e9-86c2-668103c9f205.png)

Reply:
Both `ShellHandRayPointer` and `SpherePointer` use the `BaseControllerPointer` implementation of `Position`:
https://github.com/microsoft/MixedRealityToolkit-Unity/blob/6081dcbb5f2eb1e89e26d2064e61920809bdbbb6/Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs#L374

Neither pointers define a value for raycastOrigin in their prefabs, so this just returns the pointers transform.

The transform appears to be controlled by ControllerPoseSynchronizer (from which `BaseControllerPointer` inherits). The [`poseAction`](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/6081dcbb5f2eb1e89e26d2064e61920809bdbbb6/Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ControllerPoseSynchronizer.cs#L83) inspector option seems to determine where on the controller this ends up being.

I guess if we were to use the pointer's `Rays` property, we could access this value here, but will it work for all pointers? Currently what we're doing here works grand.

Reply:
For near interaction, the origin of the grab point should be the average of the index and thumb, I was checking this earlier but didn't notice that the average calculation was commented out, great catch @keveleigh. 

It would be really awesome if we could fix the sphere center computation from palm to average (index, thumb) before our GA release. @MenelvagorMilsom are you able to fix this, since I'm OOF?

Reply:
@wiwei @thalbern and @cre8ivepark  FYI - it looks like we may be using the wrong position when doing hit testing for near interaction. It seems like we are using the index tip position when doing our sphere cast, when we should be using the average of index and thumb.

For doing actual movement / manipulation, using the palm (controller position) is a more stable point than the index and thumb, which is why we are having the transform controlled by ControllerPoseSynchronizer. 

However **the hit tests for computing which object is in focus / should receive input** should be done via average of index and thumb, to match the shell.

Reply:
Ahh interesting. Since we use `pointerPosition` as the origin of the raycast due to its use in updating `Rays[0]`, it feels like we might have a missed setting or something somewhere to update the position to the same value.

>but will it work for all pointers

It should! Hopefully all pointers have their rays updated properly to where they should be raycasting from (which feels like it should be their position?).

>Currently what we're doing here works grand.

Sounds good!

Reply:
Created #5450 and will look into it soon.